src/lib.rs

```diff
@@ -310,18 +354,26 @@ pub fn apply_filter(bpf_filter: BpfProgramRef) -> Result<()> {
     };
     let bpf_prog_ptr = &bpf_prog as *const sock_fprog;
 
+    // Until https://github.com/rust-lang/libc/issues/3342 is fixed, define locally
+
     // SAFETY:
     // Safe because the kernel performs a `copy_from_user` on the filter and leaves the memory
     // untouched. We can therefore use a reference to the BpfProgram, without needing ownership.
     let rc = unsafe {
-        libc::prctl(
-            libc::PR_SET_SECCOMP,
-            libc::SECCOMP_MODE_FILTER,
+        libc::syscall(
+            libc::SYS_seccomp,
+            SECCOMP_SET_MODE_FILTER,
+            flags,
             bpf_prog_ptr,
         )
     };
-    if rc != 0 {
-        return Err(Error::Prctl(io::Error::last_os_error()));
+
+    // Per manpage, if TSYNC fails, retcode is >0 and equals the pid of the thread that caused the
+    // failure. Otherwise, error code is -1 and errno is set.
+    if rc == -1 {
```

Comment:
let's add `rc < 0` here. Otherwise, if the kernel were to return a negative value, we would return `Ok`